diff --git a/src/jsgrid.core.js b/src/jsgrid.core.js
index 1138881f..6c2dc98d 100755
--- a/src/jsgrid.core.js
+++ b/src/jsgrid.core.js
@@ -1452,7 +1452,7 @@
 
             return this._controllerCall("updateItem", updatedItem, args.cancel, function(loadedUpdatedItem) {
                 var previousItem = $.extend(true, {}, updatingItem);
-                updatedItem = loadedUpdatedItem || $.extend(true, updatingItem, editedItem);
+                updatedItem = loadedUpdatedItem || $.extend(true, updatingItem, updatedItem);
 
                 var $updatedRow = this._finishUpdate($updatingRow, updatedItem, updatingItemIndex);
 
