diff --git a/src/i18n/es.js b/src/i18n/es.js
index e8fd7fe..aea05ee 100644
--- a/src/i18n/es.js
+++ b/src/i18n/es.js
@@ -4,11 +4,11 @@
         grid: {
             noDataContent: "No encontrado",
             deleteConfirm: "¿Está seguro?",
-            pagerFormat: "Paginas: {first} {prev} {pages} {next} {last} &nbsp;&nbsp; {pageIndex} de {pageCount}",
+            pagerFormat: "Páginas: {first} {prev} {pages} {next} {last} &nbsp;&nbsp; {pageIndex} de {pageCount}",
             pagePrevText: "Anterior",
             pageNextText: "Siguiente",
             pageFirstText: "Primero",
-            pageLastText: "Ultimo",
+            pageLastText: "Último",
             loadMessage: "Por favor, espere...",
             invalidMessage: "¡Datos no válidos!"
         },
